pkg/reconciler.v1/common/service.go

```diff
@@ -72,40 +72,40 @@ func (r *KubeflowServiceReconciler) OverrideForServiceInterface(ui ReconcilerUti
 }
 
 // GetPortsFromJob gets the ports of job container. Port could be nil, if distributed communication strategy doesn't need and no other ports that need to be exposed.
-func (r *KubeflowServiceReconciler) GetPortsFromJob(spec *commonv1.ReplicaSpec) (map[string]int32, error) {
+func (r *ServiceReconciler) GetPortsFromJob(spec *commonv1.ReplicaSpec) (map[string]int32, error) {
 	defaultContainerName := r.GetDefaultContainerName()
 	return core.GetPortsFromJob(spec, defaultContainerName)
 }
 
 // GetServicesForJob returns all services associated with this job
-func (r *KubeflowServiceReconciler) GetServicesForJob(ctx context.Context, job client.Object) ([]*corev1.Service, error) {
+func (r *ServiceReconciler) GetServicesForJob(ctx context.Context, job client.Object) ([]*corev1.Service, error) {
 	svcList := &corev1.ServiceList{}
 	err := r.List(ctx, svcList, client.MatchingLabels(r.GenLabels(job.GetName())))
 	if err != nil {
 		return nil, err
 	}
 
 	var svcs []*corev1.Service = nil
-	for _, svc := range svcList.Items {
```

Comment:
em. what's the problem of original way?

Reply:
It seems the old way will not give a new address for `svc` so when we take the pointer of `svc`, it remains the same. Please see the screenshot below.

<img width="1124" alt="Screen Shot 2021-12-12 at 20 14 53" src="https://user-images.githubusercontent.com/46096162/145711814-b384a54f-061c-45ce-93b4-e53e5a585470.png">

Reply:
I see. yeah.  This pointer points to the shared memory location that each of the slice items are being copied into.
All the pointers created using `&svc` point to the same memory location.

Current way looks good to me.